### ISSUE TITLE
segfault in ompi_mpiinfo_init on master under certain conditions

### ISSUE THREAD

**Issue by hppritcha** [open-mpi/ompi#8294]
Turns out that with master at  3f6877b5, ompi_mpiinfo_init segfaults while trying to set the ```host``` key's value when one sets OMPI_MCA_pml=XXX or uses --mca pml XXX on the mpirun command line.

I'm observing this in a slurm environment.  For some reason, the PMIX_HOSTNAME value is not being retrieved in ompi_rte_init.

PMIx returns 
```
PMIX_HOSTNAME fetch returned -46 NOT-FOUND
```
However, there's really a more basic problem.  Within ompi_mpiinfo_init is not really doing the ```right``` thing in setting the host key value.  It should be using the opal_gethostname function.

**Comment by hppritcha** [comment]
Did more digging into the PMIX_HOSTNAME issue.  Turns out its connected to using

--prtemca ras_base_launch_orted_on_hn true 

or not.  On Cray systems this mca parameter is set to true by default to allow for access to the HSN.  
If I set this parameter to false, then the pmix clients successfully retrieve the PMIX_HOSTNAME value, if true, get PMIX not found error.

Yes the mca parameter is in fact ras_base_launch_orted_on_hn and not ras_base_launch_prted_on_hn
